rajawali/src/main/java/c/org/rajawali3d/core/AFrameDelegate.java

```diff
@@ -0,0 +1,234 @@
+package c.org.rajawali3d.core;
+
+import c.org.rajawali3d.annotations.RenderThread;
+import c.org.rajawali3d.scene.AScene;
+import c.org.rajawali3d.sceneview.ASceneView;
+import org.rajawali3d.animation.Animation;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+
+import net.jcip.annotations.GuardedBy;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Abstract base class for {@link AScene} and {@link ASceneView}
+ *
+ * @author Randy Picolet
+ */
+
+public abstract class AFrameDelegate extends ACoreComponent implements FrameDelegate {
+
+    @GuardedBy("frameStartCallbacks")
+    private final List<FrameCallback> frameStartCallbacks;
+
+    @GuardedBy("frameEndCallbacks")
+    private final List<FrameCallback> frameEndCallbacks;
+
+    @GuardedBy("animations")
+    private final List<Animation> animations;
+
+    protected RenderStatus renderStatus;
+
+    private volatile boolean isEnabled;
+
+    public AFrameDelegate() {
+        super();
+        frameStartCallbacks = Collections.synchronizedList(new ArrayList<FrameCallback>());
+        frameEndCallbacks = Collections.synchronizedList(new ArrayList<FrameCallback>());
+        animations = Collections.synchronizedList(new CopyOnWriteArrayList<Animation>());
+    }
+
+    //
+    // FrameDelegate methods
+    //
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onAddToRenderControl(@NonNull RenderStatus renderStatus) {
+        this.renderStatus = renderStatus;
+        setRenderThread();
+    }
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onRemoveFromRenderControl() {
+        renderStatus = null;
+        unsetRenderThread();
+        clearQueuedTasks();
+        synchronized (frameStartCallbacks) {
+            frameStartCallbacks.clear();
+        }
+        synchronized (frameEndCallbacks) {
+            frameEndCallbacks.clear();
+        }
+        synchronized (animations) {
+            animations.clear();
+        }
+    }
+
+    @Override
+    public boolean isAttachedToRenderControl() {
+        return renderStatus != null;
+    }
+
+    @Override
+    public final void setEnabled(boolean isEnabled) {
+        this.isEnabled = isEnabled;
+    }
+
+    @Override
+    public final boolean isEnabled() {
+        return isEnabled;
+    }
+
+    @Override
+    public boolean addFrameCallback(@NonNull final FrameCallback callback) {
+        final RenderTask task = new RenderTask() {
```

Comment:
Internally I would like to see concrete implementations vs anonymous implementations whenever possible. In the examples I don't care too much but in the internal engine I think it would benefit us to be as concise as possible and anonymous classes add a lot of noise.